docs/source/app-dev/code-snippets/Types.daml

```diff
@@ -7,26 +7,30 @@ module Types where
 import Daml.Script
 import DA.Date
 
-data MyProductType = MyProductType {
-  intField: Int;
-  textField: Text;
-  decimalField: Decimal;
-  boolField: Bool;
-  partyField: Party;
-  timeField: Time;
-  listField: [Int];
-  contractIdField: ContractId SomeTemplate
-}
-
-data MySumType = MySumConstructor1 Int |
-                 MySumConstructor2 (Text, Bool)
+-- PRODUCT_TYPE_DEF_BEGIN
+data MyProductType = MyProductType with
+  intField : Int
+  textField : Text
+  decimalField : Decimal
+  boolField : Bool
+  partyField : Party
+  timeField : Time
+  listField : [Int]
+  contractIdField : ContractId SomeTemplate
+-- PRODUCT_TYPE_DEF_END
+
+-- SUM_TYPE_DEF_BEGIN
```

Comment:
I somewhat wonder if we should just talk about record & variant types instead of product & sum. Not sure the latter terms add that much value here and we use the former everywhere else. Not for this PR though.

Reply:
I think consistency would definitely be a good thing here. (And agree it's out of scope for this PR.)